pkg/network/node/networkpolicy.go

```diff
@@ -252,21 +252,32 @@ func (np *networkPolicyPlugin) syncNamespace(npns *npNamespace) {
 	}
 }
 
+func (np *networkPolicyPlugin) syncNamespaceImmediately(npns *npNamespace) {
+	otx := np.node.oc.NewTransaction()
+	np.syncNamespaceFlows(otx, npns)
+	if err := otx.Commit(); err != nil {
+		utilruntime.HandleError(fmt.Errorf("Error syncing OVS flows for namespace %q: %v", npns.name, err))
+	}
+}
+
 func (np *networkPolicyPlugin) syncFlows() {
 	np.lock.Lock()
 	defer np.lock.Unlock()
 
+	otx := np.node.oc.NewTransaction()
 	for _, npns := range np.namespaces {
 		if npns.dirty {
-			np.syncNamespaceFlows(npns)
+			np.syncNamespaceFlows(otx, npns)
 			npns.dirty = false
 		}
 	}
+	if err := otx.Commit(); err != nil {
+		utilruntime.HandleError(fmt.Errorf("Error syncing OVS flows: %v", err))
+	}
 }
 
-func (np *networkPolicyPlugin) syncNamespaceFlows(npns *npNamespace) {
+func (np *networkPolicyPlugin) syncNamespaceFlows(otx ovs.Transaction, npns *npNamespace) {
```

Comment:
This function doesn't actually sync anymore, probably should be renamed to generateNamespaceFlows or something like that instead of syncNamespaceFlows